### ISSUE TITLE
archive project

### ISSUE THREAD

**Issue by mykmelez** [mozilla/serviceworker-cookbook#312]
@digitarald Over in https://github.com/mozilla/wp-web-push/issues/337, @marco-c and I discussed archiving this project, since no one is maintaining it, and its current status is misleading (people are still filing issues on it that'll never be addressed).

Do you agree that we should archive this project?

**Comment by digitarald** [comment]
@mykmelez +1. The last maintainers moved on, so it seems the time has come.

Next steps:

- [ ] Archived status on domain or redirect

**Comment by mykmelez** [comment]
@digitarald Do you know who controls the domain? The `whois` utility only shows the registrar (Serbian National Internet Domain Registry) as the contact.

**Comment by digitarald** [comment]
@mykmelez I do. What should I do? I expected we put the archived status on gh-pages.

**Comment by mykmelez** [comment]
Ah, that makes sense (put the archived status on the pages in the gh-pages branch). Although I don't see a gh-pages branch in this repo. Perhaps the pages get generated from the master branch? I'm unsure how the site is structured, but if there's an HTML template from which the pages get generated, then I'd put a message like the following into it:

> This site has been archived and is no longer being updated. The information on this site may be out-of-date. To reuse its contents, see https://github.com/mozilla/serviceworker-cookbook.

Then, in the README for the repository, note that the project has been archived, and its code is available for reuse by forking it to create a new site. (I'm unsure I would offer to let someone reuse the domain, as it may be associated with the Mozilla brand, in which case reuse would raise a bunch of tricky questions.)

Finally, archive the repository.

**Comment by mykmelez** [comment]
After thinking a bit more, it occurs to me that the MDN team might care about this site, since MDN links to it from a variety of pages:

https://developer.mozilla.org/en-US/docs/Web/Apps/Progressive
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker
https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/JavaScript
https://developer.mozilla.org/en-US/docs/Web/API/Push_API
https://developer.mozilla.org/en-US/docs/Tools/about:debugging
https://developer.mozilla.org/en-US/docs/Web/API/Cache

On the other hand, MDN has some pwa-examples that are more recently updated, so perhaps they'd rather maintain and enhance those than take on responsibility for this site:

https://github.com/mdn/pwa-examples

@chrisdavidmills do you have any thoughts on the archival of this repository (and its website)?

**Comment by mykmelez** [comment]
@chrisdavidmills There's no hard and fast deadline. I archive unmaintained projects as they come to my attention, and this one happened to do so last month. But there's no need to archive this one by a particular deadline, unless perhaps @digitarald wants to transfer the domain name and hosting to someone else before the registration expires on November 16.

Nevertheless, we do see periodic issues filed against the repository, and I've also been pinged on Slack recently about an issue with one of the examples. I wonder if it's worth archiving the repository (while leaving the site running) with a message that we're transitioning the content to MDN (so folks don't get the wrong idea and a bad experience when filing issues).

Alternately, we could move the repo from mozilla/serviceworker-cookbook to a repo in the mdn org, such as mdn/serviceworker-examples (per the naming scheme for other "examples" repos).

**Comment by chrisdavidmills** [comment]
@mykmelez cool. Well, I'd be more than happy to have you transfer the domain to mdn/serviceworker-examples. Then we can get it fixed up as a proper MDN repo, and think about how we'd document these examples in the actual site content.

**Comment by mykmelez** [comment]
@digitarald Are you ok with transferring this repo from mozilla/serviceworker-cookbook to mdn/serviceworker-examples and giving @chrisdavidmills admin access to it, so he and other MDN folks can take over development and maintenance of the repo and use it in MDN as they see fit?

**Comment by digitarald** [comment]
@mykmelez yes, I'll coordinate with @chrisdavidmills !

**Comment by chrisdavidmills** [comment]
So, my plan here is to move the repo over to underneath the mdn GitHub org, and then present it as a set of cookbook recipes, in the say way that we do for https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook.

This is being the case, it'd be more of a set of static docs pages pointing to live examples on gh-pages. I'm guessing there's code we could probably remove that deals with generating the serviceworke.rs site (and we could just put a note on that domain saying we've moved to MDN). Would that work for you folks?

**Comment by digitarald** [comment]
> This is being the case, it'd be more of a set of static docs pages pointing to live examples on gh-pages.

Some examples have dynamic server aspects, like push and sync; but also other examples that generate "fresh" images like [this one](https://github.com/mozilla/serviceworker-cookbook/blob/master/strategy-cache-update-and-refresh/server.js).

**Comment by malcolmocean** [comment]
Is this why the site is down? I think it should display something a bit more obvious...

https://github.com/mozilla/serviceworker-cookbook/issues/314

**Comment by mykmelez** [comment]
> Is this why the site is down? I think it should display something a bit more obvious...

No, I don't think anything has been done here yet that would cause the site to go down. However, I previously noted that the domain name registration expires on November 16, and I don't find it anymore:

```
> dig serviceworke.rs

; <<>> DiG 9.10.6 <<>> serviceworke.rs
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 41277
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;serviceworke.rs.		IN	A

;; Query time: 124 msec
;; SERVER: 192.168.43.99#53(192.168.43.99)
;; WHEN: Mon Nov 19 09:02:33 PST 2018
;; MSG SIZE  rcvd: 33
```

So my suspicion is that the registration has expired.

**Comment by TylerRick** [comment]
Looks like serviceworke.rs domain has expired again. Is there any update on migrating this to MDN?

I would suggest/ask:
- rename this issue to "Migrate content to MDN"
- post a link to the content's new home :-)

**Comment by digitarald** [comment]
@chrisdavidmills any more thoughts on this? I can pay another round of serviceworke.rs to renew it.

**Comment by chrisdavidmills** [comment]
@digitarald hello mate, good to hear from you! So if you could be so kind, can you pay for another round, and then this year we'll make bloomin' sure that we migrate the thing to MDN. Seems like a good project to work on, but I just need to find the time, or find someone willing to put some time in to it.

**Comment by digitarald** [comment]
1y `.rs` domain on the house: 
![image](https://user-images.githubusercontent.com/8599/99843976-4538c780-2b27-11eb-9c7d-0992eae015af.png)

**Comment by marco-c** [comment]
@digitarald thanks!

**Comment by Rumyra** [comment]
Following up on this - are we still able to move the repo under mdn? The domain looks like it's expired again and I think it'd be good to take this opportunity to move it over and we (mdn team) can go from there - thanks!
